src/canvas_classes/tool.class.ts

```diff
@@ -1,23 +1,25 @@
+import { colors } from 'styles/colors'
+
 import { SocketApp } from 'types/socket'
 
 export class Tool {
-  protected canvas
+  protected id
   protected ctx
   protected width = 1200
   protected height = 550
   protected socket
-  protected id
-  static strokeStyle = '#000'
-  static fillStyle = '#000'
+  protected canvas
   static lineWidth = 1
+  static strokeStyle = colors.black
+  static fillStyle = colors.black
 
-  constructor(canvas: React.MutableRefObject<HTMLCanvasElement>, socket: SocketApp, id: string) {
+  constructor(canvas: HTMLCanvasElement, socket: SocketApp, id: string) {
     this.canvas = canvas
     this.socket = socket
     this.id = id
-    this.ctx = canvas.current.getContext('2d')
-    this.width = canvas.current.width
-    this.height = canvas.current.height
+    this.ctx = canvas.getContext('2d')
+    this.width = canvas.width
+    this.height = canvas.height
```

Comment:
Do you need store `ctx`, `width` and `heaight` when you store full `canvas` object?

Reply:
I left only **ctx**  not to use canvas.getContext('2d') and if(canvas.getContext('2d'))